src/routes/delegates/delegates.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { IDelegateRepository } from '../../domain/delegate/delegate.repository.interface';
+import { Delegate } from './entities/delegate.entity';
+import { Page } from '../common/entities/page.entity';
+import {
+  cursorUrlFromLimitAndOffset,
+  PaginationData,
+} from '../common/pagination/pagination.data';
+
+@Injectable()
+export class DelegatesService {
+  constructor(
+    @Inject(IDelegateRepository)
+    private readonly repository: IDelegateRepository,
+  ) {}
+
+  async getDelegates(
+    chainId: string,
+    routeUrl: Readonly<URL>,
+    safe?: string,
+    delegate?: string,
+    delegator?: string,
+    label?: string,
+    paginationData?: PaginationData,
+  ): Promise<Page<Delegate>> {
+    const delegates = await this.repository.getDelegates(
+      chainId,
+      safe,
+      delegate,
+      delegator,
+      label,
+      paginationData?.limit,
+      paginationData?.offset,
+    );
+
+    const nextURL = cursorUrlFromLimitAndOffset(routeUrl, delegates.next);
+    const previousURL = cursorUrlFromLimitAndOffset(
+      routeUrl,
+      delegates.previous,
+    );
+
+    return <Page<Delegate>>{
+      count: delegates.count,
+      next: nextURL?.toString(),
+      previous: previousURL?.toString(),
+      results: delegates.results,
+    };
+  }
```

Comment:
We should consider returning the OpenApi entity at this level. Currently it works because the interface matches.

For this we would need to add a constructor to the Page entity on the route level (which should be done a in a separate PR).

Reply:
My two cents: before returning OpenApi entities from services, I think we should think if _services_ do really belong to the _routes_ layer. I think we can all agree OpenApi definitions belong to _routes_ (they're REST-specific), but do the _services_ belong to _routes_? As I said in a previous PR, if we integrate another presentation technology (websockets, cli, gRPC, whatever) I'd expect to have _services_ available in _domain_ 🤔

Reply:
Services do the "business" logic associated with the route ie.: presentation layer. The term itself shouldn't be a big concern – it was taken from NestJS.

If we move the Service to the domain:
1.  What is the role of the service in the domain?
2. How do you encapsulate business logic for the route?

Reply:
After an offline talk, I think both of us agree this possible refactor is not needed right now, and we should focus in having feature parity with the current CGW. I'll clarify my concerns/proposal in the future 🙂 thanks.